addons/controllers/clusterbootstrap_controller.go

```diff
@@ -132,7 +132,8 @@ func (r *ClusterBootstrapReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, err
 	}
 
-	tkrName := util.GetTKRNameForCluster(r.context, r.Client, cluster)
+	// TODO: What about those classy clusters that aren't labeled with any TKR related labels?
+	tkrName := util.GetTKRNameForCluster(r.context, r.Client, cluster, constants.TKRLabelClassyClusters)
```

Comment:
Please get the label directly. By this point we should have filtered out any clusters that are not carrying the `run.tanzu.vmware.com/tkr` label.
 
```suggestion
	tkrName := cluster.Labels[constants.TKRLabelClassyClusters]
```